src/tickets/tickets.routes.ts

```diff
@@ -0,0 +1,123 @@
+import { FastifyInstance } from 'fastify'
+import {
+  ticketSchema,
+  ticketSchemaWithAttachments,
+} from './schema/ticket.schema'
+import { errorSchema } from '../users/schema/error.schema'
+import {
+  deleteTicketHandler,
+  getTicketHandler,
+  getTicketsHandler,
+  patchTicketHandler,
+  postTicketHandler,
+} from './tickets.service'
+import { responseSchema } from '../schema/response.schema'
+import { transformResponse } from '../utils/transformResponse'
+
+const ticketsRoutes = (
+  fastify: FastifyInstance,
+  _: object,
+  done: () => void
+) => {
+  fastify.get(
+    '/tickets',
+    {
+      schema: {
+        response: {
+          200: responseSchema(ticketSchemaWithAttachments),
+          401: errorSchema,
+        },
+      },
+      preHandler: fastify.auth([fastify.isUserLoggedIn]),
+    },
+    async (_, reply) => {
+      const tickets = await getTicketsHandler(fastify)
+      const transformedResponse = transformResponse(tickets)
+
+      reply.send(transformedResponse)
+    }
+  )
+
+  fastify.get(
+    '/tickets/:uuid',
+    {
+      schema: {
+        response: {
+          200: responseSchema(ticketSchema),
+          404: errorSchema,
+        },
+      },
+      preHandler: fastify.auth([fastify.isUserLoggedIn]),
+    },
+    async (request, reply) => {
+      const { uuid } = request.params as { uuid: string }
+
+      const ticket = await getTicketHandler(fastify, uuid)
+
+      const transformedResponse = transformResponse(ticket)
+
+      reply.send(transformedResponse)
+    }
+  )
+
+  fastify.post(
+    '/tickets',
+    {
+      schema: {
+        response: {
+          200: responseSchema(ticketSchema),
+          404: errorSchema,
+        },
+      },
```

Comment:
can't request schema be used here? you wouldn't have to check it in handlers

Reply:
Good point. There is possible body param in schema validation.